src/test/java/com/booleanuk/extension/Extension3Test.java

```diff
@@ -4,9 +4,14 @@
 import org.junit.jupiter.api.Test;
 
 public class Extension3Test {
-    @Test
-    public void shouldPass() {
-        Poker poker = new Poker();
-        Assertions.assertEquals("Replace this method with your own methods and tests", poker.extensionThreeMethods());
-    }
+
+//    @Test
+//    public void compareThreeHands() {
+//        Poker poker = new Poker();
+//        String[][] hands = {new String[]{"2 Heart", "2 Clove", "2 Diamond", "9 Diamond", "3 Diamond"},
+//                new String[]{"Q Clove", "9 Heart", "3 Heart", "6 Diamond", "2 Heart"},
+//                new String[]{"Q Diamond", "Q Heart", "4 Heart", "5 Diamond", "7 Heart"}};
+//        String[] result = poker.extensionThreeMethods(hands);
+//        Assertions.assertArrayEquals(new String[]{"2 Heart", "2 Clove", "2 Diamond", "9 Diamond", "3 Diamond"}, result);
+//    }
 }
```

Comment:
Overall, looks good. You could consider optimizing the code by reducing redundancy, especially in the hand evaluation logic if you wanted to improve it.